### DISCUSSION TITLE
channel  message  text search slow

### DISCUSSION THREAD

**Discussion by ourlang** [nextgenhealthcare/connect#6065] [Q&A]
Searching for a text match takes 30 minutes

Is there any optimization method or is there something wrong with the operation

<img width="463" alt="1705589430716" src="https://github.com/nextgenhealthcare/connect/assets/32355185/49b95897-39fb-415c-8b7e-3bc693799c87">

**Comment by pacmano1** [comment]
The way you are searching requires a full table scan.  Mirth will search everywhere (source and destinations).    So the search time will be impacted by the number of messages you have and from your message IDs it looks like you have millions of messages.

For faster searching, create and use metedata(see the advanced button).   You can also limit dates.

Lastly, your question is not an "issue", it is a something you do not undertand about Mirth.  Please do not create issues unless it is confirmed bug.

**Reply by pacmano1** [reply]
Your issue has been converted to a discussion.

**Comment by jonbartels** [comment]
Your current message ID is over 4 million. Is pruning turned on for this channel?
